sdk/core/azure-core/src/http/curl/curl_websockets.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/policies/policy.hpp"
+#include "azure/core/http/transport.hpp"
+#include "azure/core/http/websockets/curl_websockets_transport.hpp"
+#include "azure/core/internal/diagnostics/log.hpp"
+#include "azure/core/platform.hpp"
+
+// Private include
+#include "curl_connection_private.hpp"
+
+#if defined(AZ_PLATFORM_POSIX)
+#include <poll.h> // for poll()
+#include <sys/socket.h> // for socket shutdown
+#elif defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+#include <winapifamily.h>
+#include <winsock2.h> // for WSAPoll();
+#endif
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+
+  void CurlWebSocketTransport::NativeClose() { m_upgradedConnection->Shutdown(); }
```

Comment:
Is Shutdown safe to be called multiple times?  If by some unknown reason NativeClose() gets called multiple times...

Reply:
CurlNetworkConnection::Shutdown() simply sets m_isOpen to false.
WinHttpNetworkConnection::Shutdown() calls m_socketHandle.reset().

Both of those should be safe to be called multiple times.